custom_components/ha-google-home/__init__.py

```diff
@@ -46,64 +45,50 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     password = entry.data.get(CONF_PASSWORD)
     android_id = entry.data.get(CONF_ANDROID_ID)
 
-    session = async_get_clientsession(hass)
-    client = GlocaltokensApiClient(username, password, session, android_id)
+    session = async_get_clientsession(hass, verify_ssl=False)
+
+    glocaltokens_client = GlocaltokensApiClient(
+        hass, username, password, session, android_id
+    )
+    zeroconf_instance = await zeroconf.async_get_instance(hass)
+
+    async def async_update_data():
+        return await glocaltokens_client.get_google_devices_information(
+            zeroconf_instance
+        )
```

Comment:
Why not using lamda here? I think it's cleaner than defining sub function in the middle.

Reply:
You are right! Haven't worked with lambda much before. Do you mind sharing a snippet how you'd do that? :)

Reply:
There are few tricky parts here.
1. There are no async lamdas in python. But you can do
```
async_update_data = lamda: glocaltokens_client.get_google_devices_information(zeroconf_instance)
```
It will return awaitable as data coordinator expects.
2. PEP8 discourages using named lamdas (assigning lamda to variable): https://www.python.org/dev/peps/pep-0008/#programming-recommendations

So there are 2 options here:
1. Inline lamda in `DataCoordinator` constructor.
2. Move `async_update_data()` to the top of `async_setup_entry()`, so it's a bit better separated from the rest of the function code. It's probably worth moving `zeroconf_instance` fetching as well to have it before `async_update_data` definition, but it should work even without this.

Reply:
Actually I see this is quite a common practice in home assistant components to use `async def async_update_data()`, probably it's better to keep it this way, may only move a bit.

Reply:
Awesome, thanks! 💥